ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/context/AppContextImpl.java

```diff
@@ -481,7 +490,10 @@ public ActiveRuntime getActiveRuntime() {
 
     @Override
     public void onWindowClosing(WindowActionEvent event) {
-      appStateManager.get().persistWorkspaceState(getWorkspaceId());
+      Workspace workspace = getWorkspace();
```

Comment:
What reason call getWorkspace() here?

Reply:
Because, `onWindowClosing` may be called when workspace has stopped. And calling `getWorkspaceId` will throw an exception, because current workspace if not provided. So we'll be on the safe side, we'll check if we have a workspace as initialized object.